app/src/main/java/fr/gaulupeau/apps/Poche/service/SecondaryService.java

```diff
@@ -243,26 +244,52 @@ private void fetchImages(ActionRequest actionRequest) {
         }
 
         ArticleDao articleDao = getDaoSession().getArticleDao();
-        List<Article> articleList = articleDao.queryBuilder()
+
+        QueryBuilder<Article> queryBuilder = articleDao.queryBuilder()
                 .where(ArticleDao.Properties.ImagesDownloaded.eq(false))
-                .orderAsc(ArticleDao.Properties.ArticleId).list(); // TODO: lazyList
+                .orderAsc(ArticleDao.Properties.ArticleId);
+
+        int totalNumber = (int)queryBuilder.count();
+        Log.d(TAG, "fetchImages() total number: " + totalNumber);
+
+        if(totalNumber == 0) {
+            Log.d(TAG, "fetchImages() nothing to do");
+            return;
+        }
+
+        int dbQuerySize = 50;
+
+        queryBuilder.limit(dbQuerySize);
 
-        List<Integer> updatedArticles = new ArrayList<>(articleList.size());
+        List<Integer> updatedArticles = new ArrayList<>(totalNumber);
 
-        Log.d(TAG, "fetchImages() articleList.size()=" + articleList.size());
-        int i = 0, totalNumber = articleList.size();
-        for(Article article: articleList) {
-            Log.d(TAG, "fetchImages() processing " + i++ + ". articleID=" + article.getArticleId());
-            postEvent(new FetchImagesProgressEvent(actionRequest, i, totalNumber));
+        int offset = 0;
 
-            ImageCacheUtils.cacheImages(article.getArticleId().longValue(), article.getContent());
+        while(true) {
+            Log.d(TAG, "fetchImages() looping; offset: " + offset);
 
-            updatedArticles.add(article.getArticleId());
+            List<Article> articleList = queryBuilder.list();
+
+            if(articleList.isEmpty()) break;
```

Comment:
it would be nice to have debug message in this case, otherwise you only see the debug message "started" but nothing like "finished"